ssl.go

```diff
@@ -178,7 +178,10 @@ func sni_cb_thunk(p unsafe.Pointer, con *C.SSL, ad unsafe.Pointer, arg unsafe.Po
 
 	s := &SSL{ssl: con}
 	// This attaches a pointer to our SSL struct into the SNI callback.
-	C.SSL_set_ex_data(s.ssl, get_ssl_idx(), unsafe.Pointer(s))
+	//go vet complains here:
+	//183:42: possibly passing Go type with embedded pointer to C
```

Comment:
Do we need to worry about these?

Reply:
not really -- see https://github.com/mendersoftware/openssl/pull/15 it is a workaround for the go vet, cherry picked here, for the units to pass, will be merged with https://github.com/mendersoftware/openssl/pull/15

Reply:
yesterday I created a follow up ticket [MEN-3882](https://tracker.mender.io/browse/MEN-3882) to investigate.